src/main/java/de/craften/ui/swingmaterial/MaterialWindow.java

```diff
@@ -8,45 +8,439 @@
+     */
+    public void setWindowSize(Dimension d) {
+        //contentPane.setSize(d);
+        this.setWindowSize(d.width, d.height);
+    }
+    
+    /**
+     * {@inheritDoc}
+     * <p/>
+     * <b>NOTE:</b> If this window is casting a shadow, calling this method
+     * might provide unnacurate results unless you take into account the margin
+     * constants found in {@link MaterialShadow}. In order to make things easier
+     * to handle, it is preferable to use {@link #getWindowSize()}.
+     * @return a {@link Dimension} object that indicates the size of this window
+     * @deprecated use {@link #getWindowSize()} instead.
+     */
+    @Override
+    @Deprecated
+    public Dimension getSize() {
+        return super.getSize();
+    }
+    
+    /**
+     * Alternative method to {@link JFrame#setSize(java.awt.Dimension)}, returns
+     * the size of this window in the form of a {@link Dimension} object. Use it
+     * to get the size of this window in the desktop, excluding any margins that
+     * could be present because of the shadow. The {@code height} field of the
+     * {@code Dimension} object contains this window's height, and the
+     * {@code width} field contains this window's width.
+     * @return a {@link Dimension} object that indicates the size of this window
+     * @see #setWindowSize
+     */
+    public Dimension getWindowSize() {
+        return new Dimension(getWindowWidth(), getWindowHeight());
+    }
+    
+    @Override
+    public void setExtendedState(int state) {
+        if (shadowPane != null)
+            shadowPane.setDeployed((state & Frame.MAXIMIZED_BOTH) != Frame.MAXIMIZED_BOTH);
+        super.setExtendedState(state);
+    }
+    
+    /**
+     * Use this method to check if this {@code MaterialFrame} is casting a
+     * shadow. This affects how the window is positioned.<br/>
+     * <br/>
+     * For a window to be shadowed, the current GraphicsEnvironment must support
+     * per-pixel translucency. Also, when maximized, the shadow is automatically
+     * hidden so the contents can fill the available area.
+     * @return {@code true} if this window is currently casting a shadow on the
+     *         desktop, {@code false} otherwise.
+     */
+    private boolean isShadowed() {
+        return shadowPane != null && shadowPane.isDeployed();
+    }
+    
+    /**
+     * A {@code ShadowPane} provides the space needed for the shadow of a {@link
+     * MaterialWindow} to be displayed.
+     */
+    protected class ShadowPane extends JPanel {
+        private final MaterialShadow shadow;
+        private boolean deployed;
+
         ShadowPane() {
+            shadow = new MaterialShadow();
             setLayout(new BorderLayout());
             setOpaque(false);
-            setBorder(new EmptyBorder(MaterialShadow.OFFSET_TOP, MaterialShadow.OFFSET_LEFT, MaterialShadow.OFFSET_BOTTOM, MaterialShadow.OFFSET_RIGHT));
+            setDeployed(true);
```

Comment:
Oh, I see I forgot to add docs for that. When a `ShadowPane` is "deployed", it wraps the window contents, casting a shadow around them. This doesn't happen, for instance, when the window is maximized, so `setDeployed()` is called from `MaterialWindow.setExtendedState()` to adjust this value as necessary.